FBKVOController/FBKVOController.m

```diff
@@ -268,6 +280,17 @@ - (void)observe:(id)object info:(_FBKVOInfo *)info
   
   // add observer
   [object addObserver:self forKeyPath:info->_keyPath options:info->_options context:(void *)info];
+
+  // this could happen when `NSKeyValueObservingOptionInitial` is one of the NSKeyValueObservingOptions,
+  // and the observer is unregistered within the callback block.
+  // at this time the object has been registered as an observer (in Foundation KVO),
+  // so we can safely unobserve it if we need to.
+  BOOL shouldUnobserve = info->_state == _FBKVOInfoStateNotObserving;
```

Comment:
I believe this actually could be simplified into:
```objc
if (info->_state == _FBKVOInfoStateInitial) {
  info->_state = FBKVOInfoStateObserving;
} else {
  [object removeObserver:self forKeyPath:info->_keyPath options:info->_options context:(void *)info];
}
```

It has the same guarantees, but probably is a tad faster, since you are not going to perform lock/remove/unlock?
I can merge it this way though as well, it looks fine and provides the same guarantees.